jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java

```diff
@@ -608,6 +614,21 @@ private boolean canSelectPlayer(byte playerIdOfSelected) {
 		return MatchConstants.ENABLE_ALL_PLAYER_SELECTION || playerIdOfSelected == playerId;
 	}
 
+	private void filterWounded() {
+		if(currentSelection.getSelectionType() != ESelectionType.SOLDIERS && currentSelection.getSelectionType() != ESelectionType.PEOPLE && currentSelection.getSelectionType() != ESelectionType.SPECIALISTS)
+			return;
+
+		final List<ISelectable> selected = new LinkedList<>();
+
+		for(ISelectable selectable : currentSelection) {
+			Movable movable = (Movable)selectable;
+			if(movable.getHealth() < movable.getMovableType().health)
```

Comment:
Add a new Method to ISelectable. Then use:
```
if (selectable.isWounded()) { ... }
```

Buildings always return false.

Then you can skip the test at the beginning of the method as well, since all selectables support the same interface.